lib/pure/json.nim

```diff
@@ -913,14 +913,20 @@ when defined(js):
 
   proc parseNativeJson(x: cstring): JSObject {.importjs: "JSON.parse(#)".}
 
+  proc isInteger[T](x: T): bool {.importjs: "Number.isInteger(#)".}
+  proc isSafeInteger[T](x: T): bool {.importjs: "Number.isSafeInteger(#)".}
```

Comment:
move to `std/private/jsutils`

Reply:
I'm not sure the parameters, maybe `isSafeInteger` should choose `someInteger`?

Reply:
I think it's best to keep it unrestricted, it avoids user doing an extra check (and isSafeInteger allows arbitrary input types)

also, in this or followup PR you can add this `runnableExamples` to std/jsutils :
```nim
proc isInteger[T](x: T): bool {.importjs: "Number.isInteger(#)".}
proc isSafeInteger[T](x: T): bool {.importjs: "Number.isSafeInteger(#)".}

import std/jsffi
assert not "123".toJs.isSafeInteger
assert 123.toJs.isSafeInteger
assert 9007199254740991.toJs.isSafeInteger
assert not 9007199254740992.toJs.isSafeInteger
```

* and in this or future PR, we should also add `Number.MAX_SAFE_INTEGER`

Reply:
make sense

Reply:
see https://github.com/nim-lang/Nim/pull/17355